File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/addressspace/nodes/UaReference.java
package org.opcfoundation.webserver.addressspace.nodes;

public class UaReference {
    private final UaNode linkedNode;
    private final UaReferenceType reference;
    private final boolean isForward;

    public UaReference(
            UaNode linkedNode,
            UaReferenceType reference,
            boolean isForward)
    {
        this.linkedNode = linkedNode;
        this.reference = reference;
        this.isForward = isForward;
    }

    public UaNode linkedNode() {
        return linkedNode;
    }

    public UaReferenceType reference() {
        return reference;
    }

    public boolean isForward() {
        return isForward;
    }

    @Override
    public String toString() {
        String ret = "{";
        ret += "Linked Node: " + linkedNode.nodeId() + " " + linkedNode.browseName();
        ret += " Reference " + reference().browseName();
        ret += " IsForward " + isForward;
        ret += " }";
        return ret;
    }
}

File: opcua-webservice-node/source/src/server/addressspace/nodes/UaReference.ts

import { UaNode } from './UaNode';
import type { UaReferenceType } from './UaReferenceType';

export class UaReference {
    private readonly _linkedNode: UaNode;
    private readonly _reference: UaReferenceType;
    private readonly _isForward: boolean;

    constructor(
        linkedNode: UaNode,
        reference: UaReferenceType,
        isForward: boolean) {
        this._linkedNode = linkedNode;
        this._reference = reference;
        this._isForward = isForward;
    }

    get linkedNode(): UaNode {
        return this._linkedNode;
    }

    get reference(): UaReferenceType {
        return this._reference;
    }

    get isForward(): boolean {
        return this._isForward;
    }
}
